### ISSUE TITLE
XamlReader doesn't set `AutomationProperties.Name`

### ISSUE THREAD

**Issue by ramezgerges** [unoplatform/uno#14256]
### Current behavior

Setting `AutomationProperties.Name` in `XamlReader.Load` doesn't work. When I `GetName` afterwards, I get `""`. The same exactly sample works fine in a SamplesApp sample, but not using `XamlReader.Load`.

### How to reproduce it (as minimally and precisely as possible)

```csharp
var root = (StackPanel)XamlReader.Load(
					@"<StackPanel xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation' 
                             xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'
                             xmlns:primitives='using:Microsoft.UI.Xaml.Controls.Primitives'
                             xmlns:controls='using:Microsoft.UI.Xaml.Controls'> 
                             <controls:Expander x:Name ='ExpandedExpander' AutomationProperties.Name='ExpandedExpander' IsExpanded='True' Margin='12' HorizontalAlignment='Left'>
                                <controls:Expander.Header>
                                    <Grid>
                                        <Grid.ColumnDefinitions>
                                            <ColumnDefinition />
                                            <ColumnDefinition Width='80'/>
                                        </Grid.ColumnDefinitions>
                                        <StackPanel Margin='0,14,0,16'>
                                            <TextBlock AutomationProperties.Name='test' Text='This expander is expanded by default.'  Margin='0,0,0,4' />
                                            <TextBlock Text='This is the second line of text.' />
                                        </StackPanel>
                                        <ToggleSwitch Grid.Column='1'/>
                                    </Grid>
                                </controls:Expander.Header>
                                <Button AutomationProperties.AutomationId = 'ExpandedExpanderContent'> Content </Button>
                             </controls:Expander>
                                    </StackPanel>");
```

Note that you'll probably need to use fluent styles and not the legacy styles.

### Works on UWP/WinUI

None

**Comment by MartinZikmund** [comment]
Reproduced on Skia Desktop. `AutomationProperties.Name` set in `XamlReader.Load` XAML returns `""` instead of `"TestName"`.

Repro branch: `dev/mazi/repro/14256`  
Test: `Given_XamlReader.When_XamlReader_Sets_AutomationProperties_Name`
